internal/provider/incident_schedule_resource.go

```diff
@@ -0,0 +1,512 @@
+	rotationArray, err := buildScheduleUpdatePayload(old, resp)
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to update schedule, got error: %s", err))
+		return
+	}
+
+	result, err := r.client.SchedulesV2UpdateWithResponse(ctx, old.ID.ValueString(), client.SchedulesV2UpdateJSONRequestBody{
+		Schedule: client.ScheduleUpdatePayloadV2{
+			Name:     old.Name.ValueStringPointer(),
+			Timezone: old.Timezone.ValueStringPointer(),
+			Config: &client.ScheduleConfigUpdatePayloadV2{
+				Rotations: &rotationArray,
+			},
+		},
+	})
+	if err == nil && result.StatusCode() >= 400 {
+		err = fmt.Errorf(string(result.Body))
+	}
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to update schedule, got error: %s", err))
+		return
+	}
+
+	old = r.buildModel(result.JSON200.Schedule)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &old)...)
+}
+
+func (r *IncidentScheduleResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var data *IncidentScheduleResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	_, err := r.client.SchedulesV2DestroyWithResponse(ctx, data.ID.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to delete schedule, got error: %s", err))
+		return
+	}
+}
+
+func (r *IncidentScheduleResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+func buildScheduleCreatePayload(data *IncidentScheduleResourceModel, resp *resource.CreateResponse) ([]client.ScheduleRotationCreatePayloadV2, error) {
+	rotationArray := make([]client.ScheduleRotationCreatePayloadV2, 0, len(data.Rotations))
+	for _, rotation := range data.Rotations {
+		for _, version := range rotation.Versions {
+			workingIntervals := make([]client.ScheduleRotationWorkingIntervalV2, 0, len(version.WorkingIntervals))
+			for _, workingInterval := range version.WorkingIntervals {
+				workingIntervalWeekday := client.ScheduleRotationWorkingIntervalV2Weekday(workingInterval.Day.ValueString())
+				workingIntervals = append(workingIntervals, client.ScheduleRotationWorkingIntervalV2{
+					StartTime: workingInterval.Start.ValueString(),
+					EndTime:   workingInterval.End.ValueString(),
+					Weekday:   workingIntervalWeekday,
+				})
+			}
+
+			layers := make([]client.ScheduleLayerCreatePayloadV2, 0, len(version.Layers))
+			for _, layer := range version.Layers {
+				layers = append(layers, client.ScheduleLayerCreatePayloadV2{
+					Id:   layer.ID.ValueStringPointer(),
+					Name: layer.Name.ValueString(),
+				})
+			}
+
+			handoverStartAt, err := time.Parse(time.RFC3339, version.HandoverStartAt.ValueString())
+			if err != nil {
+				resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to create schedule, got error: %s", err))
```

Comment:
These could be a bit more useful if they specified some info about the time parsing